src/parsers/init.js

```diff
@@ -0,0 +1,39 @@
+// eslint-disable-next-line max-classes-per-file
+export class BaseParser {
+    constructor(options) {
+        this.options = options;
+    }
+
+    // eslint-disable-next-line class-methods-use-this
+    parse() {
+        throw new Error("parse() is implemented in children");
+    }
+}
+
+export class MaterialParser extends BaseParser {
```

Comment:
Move to `structure.js`.

You should define the function that forms the material here in this class as something like:

```js
parseMaterial() {

   const cell = this.getCell();
   const elements = this.getElements();
   const coordinates = this.getCoordinates();
   const constraints = this.getConstraints();

   const lattice = Lattice.fromVectors({
            a: cell.cell[0],
            b: cell.cell[1],
            c: cell.cell[2],
            type: cell.type,
        });

        const basis = new ConstrainedBasis({
            elements,
            coordinates,
            units,
            type: cell.type,
            cell: lattice.vectorArrays,
            constraints,
        });

        return {
            lattice: lattice.toJSON(),
            basis: basis.toJSON(),
            name: this.data.control.title,
            isNonPeriodic: false,
        };
```

then in child classes define the corresponding functions and reuse parseMaterial through inheritance without touching

Reply:
ok